### ISSUE TITLE
Unable to install @reduxjs/toolkit on Next.js project

### ISSUE THREAD

**Issue by DGamer007** [reduxjs/redux-toolkit#4377]
I was creating a new Next.js application with `redux` example; and I encountered this issue.

![image](https://github.com/reduxjs/redux-toolkit/assets/72591998/a3489f46-b432-4d9e-8b94-e662c3436e3f)

This issue arises even when I am trying to install `@reduxjs/toolkit` in existing Next.js project with `npm install --save @reduxjs/toolkit`.

I am currently trying on Kali Linux. But I replicated this issue on Ubuntu as well. Somehow, it seems to work on Windows.

**Comment by aryaemami59** [comment]
@DGamer007 What does your `package.json` look like?

**Comment by aryaemami59** [comment]
Looks like this is happening due to `react-native` having a peer dependency of `"react": "18.2.0"` instead of `"react": "^18.2.0"` which causes an issue if you have `"react": "^18.2.0"` since the latest version of React is now 18.3.1 which is different from 18.2.0 hence why NPM throws an error due to this version mismatch. So you can either do `npm install --force` or you can change the version of `react` in your `package.json` to be `"18.2.0"` instead of `"^18.2.0"`. Same goes for `react-dom` since `react-dom` v18.3.0 has a peer dependency of `"react": "^19.0.0"` for some reason.

**Comment by DGamer007** [comment]
> Looks like this is happening due to `react-native` having a peer dependency of `"react": "18.2.0"` instead of `"react": "^18.2.0"` which causes an issue if you have `"react": "^18.2.0"` since the latest version of React is now 18.3.1 which is different from 18.2.0 hence why NPM throws an error due to this version mismatch. So you can either do `npm install --force` or you can change the version of `react` in your `package.json` to be `"18.2.0"` instead of `"^18.2.0"`. Same goes for `react-dom` since `react-dom` v18.3.0 has a peer dependency of `"react": "^19.0.0"` for some reason.
> 
> On a side note, this issue seems similar to [reduxjs/react-redux#2161](https://github.com/reduxjs/react-redux/issues/2161).

Yes, the  `--force` option did resolve the issue but I don't think it would be feasible for production steps.

**Comment by aryaemami59** [comment]
@DGamer007 This is more of an issue with React Native than it is with redux-toolkit. I did however submit a pull request to fix it  https://github.com/facebook/react-native/pull/44325.

**Comment by DGamer007** [comment]
Roger that ! Thanks.

**Comment by markerikson** [comment]
Ultimately this is an issue on the React-Redux side (and _really_ it's an NPM issue), so I'll transfer it to the React-Redux repo.
